Data/Vector/Mutable.hs

```diff
@@ -428,3 +431,23 @@ unsafeMove = G.unsafeMove
 nextPermutation :: (PrimMonad m,Ord e) => MVector (PrimState m) e -> m Bool
 {-# INLINE nextPermutation #-}
 nextPermutation = G.nextPermutation
+
+-- Conversions - Arrays
+-- -----------------------------
+
+-- | /O(n)/ Make a copy of a mutable array to a new mutable vector.
+fromMutableArray :: PrimMonad m => MutableArray (PrimState m) a -> m (MVector (PrimState m) a)
+{-# INLINE fromMutableArray #-}
+fromMutableArray marr = do
+  let !size = sizeofMutableArray marr
+  mvec <- G.basicUnsafeNew size
+  G.basicUnsafeCopy mvec (MVector 0 size marr)
+  return mvec
+
+-- | /O(n)/ Make a copy of a mutable vector into a new mutable array.
+toMutableArray :: PrimMonad m => MVector (PrimState m) a -> m (MutableArray (PrimState m) a)
+{-# INLINE toMutableArray #-}
+toMutableArray (MVector offset size marrSrc) = do
```

Comment:
Using `Data.Primitive.Array.cloneMutableArray` avoid the needless `memset` that `newArray` uses here.

Reply:
Same thing here, good catch.